beacon-chain/sync/data_columns_reconstruct.go

```diff
@@ -0,0 +1,187 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cKzg4844 "github.com/ethereum/c-kzg-4844/bindings/go"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed"
+	statefeed "github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed/state"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/peerdas"
+	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/interfaces"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/sirupsen/logrus"
+)
+
+// recoverBlobs recovers the blobs from the data column sidecars.
+func recoverBlobs(
+	dataColumnSideCars []*ethpb.DataColumnSidecar,
```

Comment:
if the column count is < 50% we should return an error early as reconstruction is not possible

Reply:
Reconstruction is not possible with the current chosen (`2`) replication factor.
Maybe one day this replication factor will change to an other value.

In the replication factor changes, we well need to:
- update the corresponding KZG library we use,
- update `reconstructDataColumns` (where there is already a check), and
- update the early return condition.

Trying to reconstruct with less than the minimum amount of data needed will make `RecoverAllCells` to return on error. (And thus, the `recoverBlobs` function return on error as well.)

Because `recoverBlobs` is a private function, only called by `reconstructDataColumns` (which include the replication factor check), my opinion is it is not necessary to add a new return early check.